diff --git a/socket.c b/socket.c
index a01ad74..f313695 100644
--- a/socket.c
+++ b/socket.c
@@ -1720,6 +1720,11 @@ SCK_RemoveSocket(int sock_fd)
   socklen_t saddr_len;
 
   saddr_len = sizeof (saddr);
+#ifdef QNX
+  // QNX getsockname doesn't null terminate sun_path
+  // 0 initializing will ensure that there is at least a null byte where the string ends
+  memset(&saddr, 0, sizeof(saddr)); 
+#endif
 
   if (getsockname(sock_fd, &saddr.sa, &saddr_len) < 0) {
     DEBUG_LOG("getsockname() failed : %s", strerror(errno));

